rust/lance-core/src/datatypes/field.rs

```diff
@@ -36,6 +41,12 @@ use crate::{
     Error, Result,
 };
 
+#[derive(Default)]
+pub struct SchemaCompareOptions {
+    pub compare_metadata: bool,
+    pub compare_dictionary: bool,
+}
```

Comment:
Could we also add a `compare_field_ids` option?

Reply:
Would this just compare the `id` or would it also look at `parent_id`?  I think I'd expect both to match or neither to match.  `parent_id` is more of a validation thing.

Reply:
I think you can just check `id`. `parent_ids` should match if implicitly if the `ids` match and you are walking the fields correctly.